pkg/registry/apps/statefulset/strategy.go

```diff
@@ -74,6 +74,10 @@ func (statefulSetStrategy) PrepareForCreate(ctx context.Context, obj runtime.Obj
 
 	statefulSet.Generation = 1
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) {
+		statefulSet.Spec.PersistentVolumeClaimDeletePolicy = nil
```

Comment:
I would argue that setting this to the default value (which is the implicit policy in previous versions) makes the API more consistent. But we have no policy around this.

@liggitt for authoritative answer.

Reply:
I set to nil as that seemed to be what was required by step 3 at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-field-in-existing-api-version.

Happy to change if I misinterpreted that.

Reply:
When the feature is disabled, we must not allow new data to be persisted to the field in etcd (so we clear it before create and update) or return data to the user in that field for existing objects (so we avoid defaulting the field). We don't want to persist alpha-level data for a feature which might never graduate to beta.

Allowing objects which already have the field set to continue to set it on update is to avoid data-loss in scenarios where a beta-level or feature-enabled server allowed the field to be populated.

Reply:
tl;dr: the current code matches the API change requirements